src/app/pet.service.ts

```diff
@@ -17,12 +17,20 @@ export class PetService {
     headers: new HttpHeaders({ 'Content-Type': 'application/json' })
   };
 
-  pets$ = this.http.get<Pet[]>(this.petsUrl)
+  private _petsData$ = new BehaviorSubject<void>(undefined);
+  apiRequest$ = this.http.get<Pet[]>(this.petsUrl)
     .pipe(
-      tap(_ => console.log('fetched pets')),
-      catchError(this.handleError<Pet[]>('fetch pets', []))
+      tap(pet => {
+        console.log('fetch pets');
+        return pet;
```

Comment:
a `tap` is a side effect. Something that happens passively. You do not need to return anything from a tap.

Reply:
I removed the return and modified the console message.